app/repositories/root_proxies_repository.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+class RootProxiesRepository
+  attr_reader :base_url, :headers
+
+  def initialize(config:)
+    @config = config
+    @base_url = "#{rmapi_url}/rm/rmaccounts/#{config.customer_id}"
+
+    @headers = {
+      'X-Api-Key': config.api_key,
+      'Content-Type': 'application/json',
+      'Accept': 'application/json'
+    }
+  end
+
+  def find!
+    request do
+      status, body = rmapi(:get, '')
+      Result.new(
+        data: to_root_proxy(body[:proxy]),
+        status: status
+      )
+    end
+  end
+
+  def update!(attrs)
+    proxy_types_list = ProxyTypesRepository.new(config: @config).where!
+    proxy_id = attrs[:proxyTypeId]
+    request do
+      root_proxy_validation = Validation::RootProxyParameters.new(proxy_id, proxy_types_list)
+      fail ValidationError, root_proxy_validation unless root_proxy_validation.valid?
+
+      payload = attrs.to_hash.deep_symbolize_keys
+
+      payload[:proxy] = { id: payload.delete(:proxyTypeId) }
```

Comment:
Here payload that we get from UI-eholdings will contain only proxy-related information, however, for RM API, proxies and custom-labels are handled by the same PUT call, so we need to make a GET, extract custom-labels and append it to payload before we pass it to RM API; else it would reset all custom-labels.

Reply:
Yep! We learned that lesson on Thursday evening. I will add the call to get the custom labels today.